### PR TITLE
ISSUE-9 Memory deployment tests

### PR DESCRIPTION
**Pull request by chibenwa**
Result of today's MOB

Depends on https://github.com/linagora/james-project/pull/4222

### CODE REVIEW

File: openpaas-james/deployment-tests/common/src/main/java/com/linagora/openpaas/deployment/CliContract.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.testcontainers.containers.GenericContainer;
+
+public interface CliContract {
+    GenericContainer<?> jamesContainer();
+
+    @Test
+    default void cliShouldWork() throws Exception {
+        jamesContainer().execInContainer("james-cli", "addDomain", "domain.tld");
+
+        assertThat(jamesContainer().execInContainer("james-cli", "ListDomains").getStdout())
```
Comment (rouazana) [inline openpaas-james/deployment-tests/common/src/main/java/com/linagora/openpaas/deployment/CliContract.java:15]: I'm not sure testing only stdout gives me enough confidence in this test. Shouldn't we test we really added the domain?

**Review by remk — APPROVED** [review]

File: openpaas-james/deployment-tests/common/src/main/java/com/linagora/openpaas/deployment/CliContract.java (hunk 1 above)
Comment (Arsnael) [reply openpaas-james/deployment-tests/common/src/main/java/com/linagora/openpaas/deployment/CliContract.java:15]: how? by checking if webadmin works, which should maybe be an other test?

Comment (chibenwa) [reply openpaas-james/deployment-tests/common/src/main/java/com/linagora/openpaas/deployment/CliContract.java:15]: I would, **please**, like to keep the deployment test counts **as low as possible** as they are really long to play.

I am unwilling to have more than a single test for the CLI (which by the way is a net gain!)

Maybe that means breaking other testing rules like single assertion.

Let me suggest:

```
    default void cliShouldWork() throws Exception {
        assertThat(jamesContainer().execInContainer("james-cli", "ListDomains").getStdout())
            .doesNotContain("domain.tld");
            
        jamesContainer().execInContainer("james-cli", "addDomain", "domain.tld");

        assertThat(jamesContainer().execInContainer("james-cli", "ListDomains").getStdout())
```

Doing asserts on return codes is an option too.

**Review by remk — APPROVED** [review]

**Review by rouazana — APPROVED** [review]

**Comment by chibenwa** [comment]
Squashed

**Pull request merged**